vllm/v1/engine/async_llm.py

```diff
@@ -281,10 +295,13 @@ async def _run_output_handler(self):
         try:
             while True:
                 # 1) Pull EngineCoreOutput from the EngineCore.
-                outputs = await self.engine_core.get_output_async()
+                outputs: EngineCoreOutputs = (
```

Comment:
One note: this is a blocking call. So if no work is done by the EngineCore, then this will wait forever. I wonder if we need some way to send logs if no work is happening like we have now

Reply:
Oh - that's good to know! I wonder if that would mean we would need to have another data flow for the stats other than piggybacking on the engine outputs.

Reply:
Looks like we are already doing `profile` as another cross-process call other than adding request, I wonder if we can extend the interface by adding a `get_stats()` call to the `MPClient` in parallel with `get_output()`.

Caller could then have an async coroutine pulling stats from the core backend without being blocked by the `await get_output_async()`